### PR TITLE
[DEVX-1033] Use playwright@1.12.3

### PR DESCRIPTION
**Pull request by FriggaHel**
Update to playwright@1.12.3

Don't support playwright configuration file currently.

### CODE REVIEW

File: src/playwright-runner.js [hunk 1]
```diff
@@ -127,76 +127,61 @@ async function run (nodeBin, runCfgPath, suiteName) {
...
-  const excludeParams = ['name', 'platform-name', 'browser-name', 'playwright-version', 'env'];
+  const projectPath = path.dirname(runCfg.path);
+  if (!fs.existsSync(projectPath)) {
+    throw new Error(`Could not find projectPath directory: '${projectPath}'`);
+  }
+
+  // Copy playwright configuration to project folder.
+  // As we curently don't support playwright configuration yet,
+  //   and set a configuration file automatically change the rootDir.
+  //   this is a workaround.
```
Comment (alexplischke) [inline src/playwright-runner.js:144]: Improved the wording a little. Also removed the `rootDir` sentence, since I didn't understand it or saw any references made in the code to that 🤔 
```suggestion
  // Copy our own playwright configuration to the project folder,
  // as we currently don't support having a user provided playwright configuration yet.
```

File: tests/unit/src/playwright-runner.spec.js [hunk 2]
```diff
@@ -69,8 +69,31 @@ describe('folio-runner', function () {
       procArgs[0] = path.basename(procArgs[0]);
       procArgs[procArgs.length - 1] = path.basename(procArgs[procArgs.length - 1]);
       spawnArgs.cwd = path.basename(spawnArgs.cwd);
-      spawnArgs.env.FOLIO_JUNIT_OUTPUT_NAME = path.basename(spawnArgs.env.FOLIO_JUNIT_OUTPUT_NAME);
-      expect([nodeBin, procArgs, spawnArgs]).toMatchSnapshot();
+      spawnArgs.env.PLAYWRIGHT_JUNIT_OUTPUT_NAME = path.basename(spawnArgs.env.PLAYWRIGHT_JUNIT_OUTPUT_NAME);
+      //), procArgs, spawnArgs]).toMatch([
```
Comment (alexplischke) [inline tests/unit/src/playwright-runner.spec.js:73]: Leftover?

File: src/playwright-runner.js (hunk 1 above)
Comment (FriggaHel) [reply src/playwright-runner.js:144]: @alexplischke  Let me try to give a more clear explanation of what I was trying to explain concerning `rootDir`.

Here is a local way to simulate behaviour from the runner on VMs.
project & runner are not the same directory.

Initially I added the `playwright.config.js` in the runner root directory and specified it using `--config`.
By doing that, `playwright test` command discards the current working directory and assume that the `rootDir` / `projectDir` is where is located the `playwright.config.js`.

Example:
```
{17:21}~/Projects/playwright-troubleshooting/project ➭ export NODE_PATH=`realpath ../runner/node_modules`
{17:21}~/Projects/playwright-troubleshooting/project ➭ ls -1 ../
project
runner
{17:21}~/Projects/playwright-troubleshooting/project ➭ ../runner/node_modules/.bin/playwright test

Running 1 test using 1 worker

  ✓ tests/test.spec.js:18:1 › [chromium] Test within PROJECT folder (7s)


  1 passed (8s)
{17:22}~/Projects/playwright-troubleshooting/project ➭ ../runner/node_modules/.bin/playwright test --config ../runner/playwright.config.js
Using config at /Users/xxx/Projects/playwright-troubleshooting/runner/playwright.config.js

Running 1 test using 1 worker

  ✓ tests/test.spec.js:18:1 › [chromium] Test within RUNNER folder (6s)


  1 passed (7s)
{17:22}~/Projects/playwright-troubleshooting/project ➭
```

As there is no option to enable video record outside of the `playwright.config.js` file, there is no other solution as creating that file in the project folder.

(And this part is not documented in playwright docs).

Is that a better explanation ?

File: tests/unit/src/playwright-runner.spec.js (hunk 2 above)
Comment (FriggaHel) [reply tests/unit/src/playwright-runner.spec.js:73]: Yup, will remove it.

File: src/playwright-runner.js [hunk 3]
```diff
@@ -127,76 +127,59 @@ async function run (nodeBin, runCfgPath, suiteName) {
   const runCfg = await loadRunConfig(runCfgPath);
   runCfg.path = runCfgPath;
   const cwd = process.cwd();
-  let defaultArgs = {
-    param: {
-      headful: process.env.SAUCE_VM ? true : false,
```
Comment (mhan83) [inline src/playwright-runner.js:132]: Aren't these default values required?

**Review by valentyn88 — APPROVED** [review]

File: src/playwright-runner.js (hunk 3 above)
Comment (FriggaHel) [reply src/playwright-runner.js:132]: This have been fixed here: https://github.com/saucelabs/sauce-playwright-runner/pull/99/files#diff-666450f3c804e34e85768192da8dae7fd947b4b14db90b48f12cea82711fcf22R141

**Comment by alexplischke** [comment]
@valentyn88 Did you happen to test tunnels with this?

File: src/playwright-runner.js (hunk 1 above)
Comment (alexplischke) [reply src/playwright-runner.js]: Ah interesting, thanks! 😄 Yeah, more clear now 👍 
It's just not very clear from the inline docs. Based on my understanding now, I'd reword it like this:
```
  // Copy our own playwright configuration to the project folder,
  // as we currently don't support having a user provided playwright configuration yet. Setting a config file
  // also automatically changes the root directory for playwright to the directory that contains the config file.
```

**Comment by FriggaHel** [comment]
> @valentyn88 or @FriggaHel, Did you happen to test tunnels with this?

@alexplischke I did not tried tunnel per-se. (Don't know if this is available with CTR).
I tried to run the bundle (on a VM) with a HTTP_PROXY/HTTPS_PROXY var set, and this is going through it.

IMHO this is ok, that would just require a real e2e test to confirm that, but there is no reason this won't be working.

**Comment by valentyn88** [comment]
> @valentyn88 or @FriggaHel, Did you happen to test tunnels with this?

Only with changes from @FriggaHel it's possible to test the new playwright-runner and tunnel. We still have this ticket in a backlog.

**Review by alexplischke — APPROVED** [review]

**Review by tianfeng92 — APPROVED** [review]

File: playwright.config.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  use: {
+    video: 'on',
```
Comment (mhan83) [inline playwright.config.js:3]: Technically video is supposed to be off in a sauce vm (https://github.com/saucelabs/sauce-playwright-runner/blob/e8ed632f976fd4e2c3a34be18479ca660805770b/src/utils.js#L79-L81). But there's a bug in the current latest version of the runner that does not respect this (if I remember correctly its because the video param is supposed to be `--param video` but the runner was formatting the arg like `--param video=true/false`). I'm guessing it hasn't blown up in sauce because playwright includes its own ffmpeg? Not totally sure.

Anyways, just wanted to point this out. Not sure it necessarily needs to be addressed here since templating this config might not be worth the trouble right now.

Comment (FriggaHel) [reply playwright.config.js:3]: @mhan83 I disabled the video recording when running in Sauce VMs !

**Review by mhan83 — APPROVED** [review]
:shipit:
